sound/soc/sof/intel/hda.c

```diff
@@ -414,11 +415,24 @@ struct snd_soc_sof_mach *hda_machine_driver_select(struct snd_sof_dev *sdev)
 
 	return mach;
 }
+
+void hda_set_mach_params(struct snd_sof_dev *sdev,
+			 struct snd_soc_acpi_mach_params *mach_params)
+{
+	struct hdac_bus *bus = sof_to_bus(sdev);
+
+	mach_params->codec_mask = bus->codec_mask;
+	mach_params->common_hdmi_codec_drv = hda_codec_use_common_hdmi;
+}
 #else
 struct snd_soc_sof_mach *hda_machine_driver_select(struct snd_sof_dev *sdev)
 {
 	return NULL;
 }
+
+void hda_set_mach_params(struct snd_sof_dev *sdev,
```

Comment:
the use of sdev is not quite right here if we want the client to make the machine driver selection?